Python/ceval.c

```diff
@@ -2376,20 +2376,19 @@ void
 PyEval_SetProfileAllThreads(Py_tracefunc func, PyObject *arg)
 {
     PyThreadState *this_tstate = _PyThreadState_GET();
-    PyInterpreterState* interp = this_tstate->interp;
+    PyInterpreterState *interp = this_tstate->interp;
 
-    _PyRuntimeState *runtime = &_PyRuntime;
-    HEAD_LOCK(runtime);
+    INTERP_THREAD_LOCK(interp);
     PyThreadState* ts = PyInterpreterState_ThreadHead(interp);
-    HEAD_UNLOCK(runtime);
+    INTERP_THREAD_UNLOCK(interp);
 
     while (ts) {
         if (_PyEval_SetProfile(ts, func, arg) < 0) {
             PyErr_FormatUnraisable("Exception ignored in PyEval_SetProfileAllThreads");
         }
-        HEAD_LOCK(runtime);
+        INTERP_THREAD_LOCK(interp);
         ts = PyThreadState_Next(ts);
-        HEAD_UNLOCK(runtime);
+        INTERP_THREAD_UNLOCK(interp);
     }
 }
```

Comment:
Oh, it seems I didn't notice this problem. In fact, the lock should be placed at the beginning and end of `while`.

Reply:
It seems that this cannot be modified because it causes the `test_getprofile_all_threads` test to fail in CI.
```
test_getprofile_all_threads (test.test_threading.ThreadTests.test_getprofile_all_threads) ... Timeout (0:10:00)!
Thread 0x000000016da97000 (most recent call first):
  File "/Users/admin/actions-runner/_work/cpython/cpython/Lib/threading.py", line 359 in wait
  File "/Users/admin/actions-runner/_work/cpython/cpython/Lib/threading.py", line 659 in wait
  File "/Users/admin/actions-runner/_work/cpython/cpython/Lib/test/test_threading.py", line 1061 in checker
  File "/Users/admin/actions-runner/_work/cpython/cpython/Lib/threading.py", line 992 in run
  File "/Users/admin/actions-runner/_work/cpython/cpython/Lib/threading.py", line 1041 in _bootstrap_inner
  File "/Users/admin/actions-runner/_work/cpython/cpython/Lib/threading.py", line 1012 in _bootstrap
Thread 0x00000001ff5acf40 (most recent call first):
```